go/arrow/array/numericbuilder.gen.go.tmpl

```diff
@@ -96,10 +96,12 @@ func (b *{{.Name}}Builder) AppendValues(v []{{or .QualifiedType .Type}}, valid [
 		panic("len(v) != len(valid) && len(valid) != 0")
 	}
 
-	b.Reserve(len(v))
-	if len(v) > 0 {
-		arrow.{{.Name}}Traits.Copy(b.rawData[b.length:], v)
+	if len(v) == 0 {
```

Comment:
I don't think this is specific to numeric builders. Note that this panics:
```
array.NewStringBuilder(memory.DefaultAllocator).AppendValues([]string{}, []bool{})
```

Reply:
good point.